modules/objc/VectorSeachSnippets.m

```diff
@@ -0,0 +1,120 @@
+//
+//  VectorSeachSnippets.m
+//
+//  Created by Vlad Velicu on 20/03/2024.
+//
+
+#import <Foundation/Foundation.h>
+#import <CouchbaseLite/CouchbaseLite.h>
+#import <NaturalLanguage/NLEmbedding.h>
+
+CBLDatabase* database;
+CBLCollection* collection;
+NLEmbedding* model;
+
+// MARK: Configuring a project to use Vector Search.
+
+/*/
+ Download Couchbase Lite iOS Obj-C and Couchbase Lite Vector Search libraries
+ Extract the xcframeworks and copy them into the App Project Directory
+ 
+ Add both libraries to the *Frameworks, Libraries and Embedded Content* to your desired target
+ */
+
+@interface VectorSearch : NSObject 
+
+@end
+
+@implementation VectorSearch
+
+- (void) createDefaultIndex {
+    NSError* error;
+    database = [[CBLDatabase alloc] initWithName:@"vectorDB" error:&error];
+    collection = [database defaultCollection: &error];
+    
+    // MARK: Create a default Vector Index Configuration
+
+    CBLVectorIndexConfiguration* config = [[CBLVectorIndexConfiguration alloc] initWithExpression: @"vector" dimensions: 300 centroids: 20];
+    
+    // Default values already set:
+    NSLog(@"%@", config.encoding); // .scalarQuantizer(type: .SQ8)
+    NSLog(@"%u", config.metric); // .euclidean
+    NSLog(@"%u", config.minTrainingSize); // 25 * 20
+    NSLog(@"%@", config.encoding); // 256 * 20
+    
+    // MARK: Set custom optional settings
+    config.encoding = [CBLVectorEncoding none];
+    config.metric = kCBLDistanceMetricCosine;
+    config.minTrainingSize = 50;
+    config.maxTrainingSize = 300;
+}
+
+// MARK: Create Vector Index with Embedding
+- (CBLQueryResultSet*) vectorIndexEmbedding {
+    NSError* error;
+    CBLVectorIndexConfiguration* config = [[CBLVectorIndexConfiguration alloc] initWithExpression: @"vector" dimensions: 300 centroids: 20];
+    [collection createIndexWithName: @"vector_index" config: config error: &error];
+    
+    model = [NLEmbedding wordEmbeddingForLanguage:@"english"];
+    [model vectorForString: @"<word>"];
+    
+    NSString* sql = @"select meta().id, word from _default.words where vector_match(vector_index, $vector, 20)";
+    CBLQuery* query = [database createQuery: sql error: &error];
+    
+    CBLQueryParameters* parameters = [[CBLQueryParameters alloc] init];
+    [parameters setValue: <vectorArray> forName: @"vector"];
+    [query setParameters: parameters];
+    
+    return [query execute: &error];
+}
+
+// MARK: Create Vector Index with Predictive Model
+- (void) vectorIndexPredictiveModel {
+    NSError* error;
+
+    model = [NLEmbedding wordEmbeddingForLanguage:@"english"];
+    [[CBLDatabase prediction] registerModel: model withName:@"WordEmbedding"];
```

Comment:
This is incorrect. The model needs to implement CBLPredictiveModel protocol. Also, we will need to sync the code snippts for all platforms. Here are two samples:

Samples:

// Color Model (I think server team has used RGB color as one of their sample apps)

```
@interface ColorModel : NSObject <CBLPredictiveModel>
@end

@implementation ColorModel

- (CBLDictionary*) predict: (CBLDictionary*)input {
   NSString* color = [input stringForKey: @"color"];
   NSArray* rgb = [RGBModel getRGBVectorForColorName: color]; // I made up the name of the model.
   
   CBLMutableDictionary* output = [[CBLMutableDictionary alloc] init];
    [output setValue: rgb forKey: @"vector"];
    return output;
}

@end
```

// Word Model 

```
@interface WordModel : NSObject <CBLPredictiveModel>
@end

@implementation WordModel

- (CBLDictionary*) predict: (CBLDictionary*)input {
   NSString* word = [input stringForKey: @"word"];
   NSArray* vector = [WordModel getVectorForWord: word]; // I made up the name of the model.
   
   CBLMutableDictionary* output = [[CBLMutableDictionary alloc] init];
    [output setValue: vector forKey: @"vector"];
    return output;
}

@end
```

Reply:
@ElliotFrancisHunter do we want/need to align with server on this?

Reply:
Aligning with server on examples is a good idea if we have customers using multiple products. I'm in favour of this choice

Reply:
Server seems to have some demo code here https://docs.couchbase.com/server/current/vector-search/create-vector-search-index-rest-api.html#example

which is related to colour. For mobile let's use the word examples as it's simpler for integration into MVP of beta docs. We can refine them over time with more detail such as the colour model

Reply:
PM suggested to me on the other day to have demo app converted to use the server's color model. I didn't have time to do it yet but the new sample app going forward will be color. It's ok to use word model as a sample now and convert it to use color later. It's not much of the change.

Reply:
will use word model for now